libcaf_core/caf/logger.cpp

```diff
@@ -248,26 +423,35 @@ logger* logger::current_logger() {
   return current_logger_ptr.get();
 }
 
-bool logger::accepts(unsigned level, std::string_view cname) {
+bool default_logger::accepts(unsigned level, std::string_view cname) {
   if (level > cfg_.verbosity)
     return false;
   return std::none_of(global_filter_.begin(), global_filter_.end(),
                       [=](std::string_view name) { return name == cname; });
 }
 
-logger::logger(actor_system& sys) : system_(sys), t0_(make_timestamp()) {
+logger::logger() {
   // nop
 }
 
 logger::~logger() {
+  // nop
+}
+
+default_logger::default_logger(actor_system& sys)
+  : t0_(make_timestamp()), system_(sys) {
+  // nop
+}
+
+default_logger::~default_logger() {
   stop();
   // tell system our dtor is done
-  std::unique_lock<std::mutex> guard{system_.logger_dtor_mtx_};
-  system_.logger_dtor_done_ = true;
-  system_.logger_dtor_cv_.notify_one();
+  std::unique_lock<std::mutex> guard{system_.logger_dtor_mtx()};
+  system_.logger_dtor_done(true);
+  system_.logger_dtor_cv().notify_one();
```

Comment:
Just a note, this setup is too complicated (has nothing to do with your changes). When the refactoring itself is done, I'll think of a better way we can do this so that people don't need to know about internals like this when subtyping `logger`.

Reply:
I had initially chosen to move these functions inside the logger destructor. Maybe, we can call some overloaded functions which can call these methods instead of calling them in the destructor directly? It still does not fix the issue completely but it may be worth refactoring for now.